javalib/src/main/scala/java/net/DatagramPacket.scala

```diff
@@ -0,0 +1,89 @@
+package java.net
+
+final class DatagramPacket(
+    var buf: Array[Byte],
+    var offset: Int,
+    var length: Int,
+    // bufLength: Int,
+    var address: InetAddress,
+    var port: Int
+) {
```

Comment:
End of sprint, so I will have to think about this one.

My first thoughts are that the various  'var' modifiers
break encapsulation or opaqueness.  That is, if one
has a variable named dp of type DatagramPacket, with
this declaration, one could write "dp.buf = null"  when
it appears that the intent of the JDK documentation
is that one use "dp.setData(somethingNotNull)"

 This class comes from Java 1.0, so I think an implementation is forced to think like 1993 or so, years before Java 1.0 was released.  That leads to code which
seems strange to contemporary eyes.

Get your barf bucket ready...

I believe that a private zero argument primary constructor
is necessary, with a series of class local variables and
a series of secondary constructors having one or more arguments.  

At worst implementation, each secondary calls the primary and fills in all the local variables.  It may be possible to declare the secondary so that each does some work and then calls another to fill in unspecified values. I get that sequence wrong all the time, so I can not give example code on the fly. I would have to do a private sandbox implementation.